src/types/connection/connection_type.rs

```diff
@@ -180,16 +219,36 @@ impl<T: OutputValueType + Send + Sync, E: ObjectType + Sync + Send> ObjectType
     async fn resolve_field(&self, ctx: &Context<'_>) -> Result<serde_json::Value> {
         if ctx.name.node == "pageInfo" {
             let ctx_obj = ctx.with_selection_set(&ctx.selection_set);
-            return OutputValueType::resolve(self.page_info().await, &ctx_obj, ctx.item).await;
+            return OutputValueType::resolve(&self.page_info, &ctx_obj, ctx.item).await;
         } else if ctx.name.node == "edges" {
             let ctx_obj = ctx.with_selection_set(&ctx.selection_set);
-            return OutputValueType::resolve(&self.edges().await, &ctx_obj, ctx.item).await;
+            let edges = self
+                .nodes
+                .iter()
+                .map(|record| {
+                    Some(Edge {
+                        cursor: &record.cursor,
+                        extra_type: &record.edge,
+                        node: &record.element,
+                    })
+                })
+                .collect_vec();
+            return OutputValueType::resolve(&edges, &ctx_obj, ctx.item).await;
         } else if ctx.name.node == "totalCount" {
             let ctx_obj = ctx.with_selection_set(&ctx.selection_set);
-            return OutputValueType::resolve(&self.total_count().await, &ctx_obj, ctx.item).await;
+            return OutputValueType::resolve(
```

Comment:
Since `total_count` isn't required by the spec, is this allowed to be excluded by an implementation?

Reply:
If we remove the `totalCount`, when the user wants this field, we have not provided a way to add any field to the Connection. 😂

Reply:
@sunli829 if we don't make this optional, we cannot support streams that end between a `cursor` and `limit` and you force the entire stream to be consumed. In large data sets, that fails.